tests/hash_test.cpp

```diff
@@ -0,0 +1,132 @@
+#include "catch/catch.hpp"
+
+#include <chrono>
+#include <functional>
+#include <unordered_set>
+
+#include "map.h"
+
+#include "enums.h"
+
+// A larger number for this would be GREAT, but the test isn't efficient enough to make it larger.
+constexpr int MAX_COORDINATE = 100;
+
+TEST_CASE( "point_hash_distribution", "[hash]" )
+{
+    std::vector<point> points = closest_points_first( MAX_COORDINATE, { 0, 0 } );
+    std::unordered_set<size_t> hashtogram;
```

Comment:
I did one better and used a std::array instead since we can calculate the number of elements at compile time.

Reply:
std::array is not better since it ends up on the stack by default, so I fell back to a std::vector as suggested, with a reserve before insertion.

Similar results are achievable by reserving the unordered_set, but the vector+sort+unique solution ended up faster given the (arbitrary-ish) goal of coming in at about 0.5s of runtime.

At this point nearly *all* the time is spent in std::sort, which seems to be slower than expected, but it's still the best option and the test does what I want, which is fail with the old, bad hash and pass with the new hash (also the zcurve hash, but that's not included).